src/gtest/test_merkletree.cpp

```diff
@@ -322,3 +323,22 @@ TEST(orchardMerkleTree, appendBundle) {
         EXPECT_EQ(newTree.root(), readBack.root());
     }
 }
+
+TEST(saplingMerkleTree, subtree_roots)
+{
+    SaplingMerkleTree tree;
+    for (size_t i = 0; i < (1 << 16); i++) {
+        EXPECT_FALSE(tree.complete_subtree_root().has_value());
+        EXPECT_EQ(tree.current_subtree_index(), 0);
+        AppendRandomLeaf(tree);
+    }
+    auto subtree_root = tree.complete_subtree_root();
+    ASSERT_TRUE(subtree_root.has_value());
+    EXPECT_EQ(tree.current_subtree_index(), 1);
```

Comment:
"current" means "will next be appended to"? Maybe this should actually be named `appendable_subtree_index` or something of the sort.